alf/algorithms/trac_algorithm.py

```diff
@@ -239,16 +227,13 @@ def _update_total_dists(new_action, exp, total_dists):
                 observation=exp.observation, step_type=exp.step_type)
             policy_step = self._ac_algorithm.predict(
                 time_step=time_step, state=state, epsilon_greedy=1.0)
-            if self._action_distribution_spec is None:
-                new_action = policy_step.action
-            else:
-                assert (
-                    common.is_namedtuple(policy_step.info)
-                    and "action_distribution" in policy_step.info._fields
-                ), ("PolicyStep.info from ac_algorithm.predict() should be "
-                    "a namedtuple containing `action_distribution` in order to "
-                    "use TracAlgorithm.")
-                new_action = policy_step.info.action_distribution
+            assert (
```

Comment:
redundant assertion, already checked in `_make_policy_step`

Reply:
That only asserts for train_step() and rollout(). Here is predict().